visualization/app/codeCharta/ui/dialog/dialog.changelog.component.ts

```diff
@@ -0,0 +1,46 @@
+import "./dialog.changelog.component.scss"
+// @ts-ignore
+import md from "../../../../../CHANGELOG.md"
+import packageJson from "../../../../package.json"
+
+export class DialogChangelogController {
+	private _viewModel: {
+		version: string
+		md: any
+	} = {
+		version: "",
+		md: null
+	}
+
+	constructor(private $mdDialog) {
+		"ngInject"
+		const changelogLines = md.split("\n")
+		// Get current version
+		this._viewModel.version = packageJson.version
+		// Get the Changelog just for the newest version
```

Comment:
If I understood correctly, we show all the new changes, not only the newest?
That would introduce a little problem. When someone hasn't used CodeCharta for a long time or uses it for the first time. The list of Changes can become long. That becomes overwhelming.

Reply:
@IhsenBouallegue ideally, the changelog is only visible to people opened CodeCharta before (as it's meant as a feature for people who already used CC). Everyone else should be fine to read about the features before starting to use it.

Reply:
We could let the user navigate through the versions instead of showing `Know more`.

Reply:
Here is a suggestion (this is not implemented yet) but I did a quick mock-up to get your feedback

- If the user opens CodeCharta for the first time, they don't see a changelog.

- If the user comes back and there are less than 5 new versions, they see the latest changes and can check also the other versions.

![CodeCharta changelog version1](https://user-images.githubusercontent.com/48621967/130053522-76b2e1c2-6653-4630-819b-1e040b55605b.PNG)

- If the user comes back and there are more than 5 new versions, the get promoted to check the changelog and only the newest 5 get shown in the dialog. 

![CodeCharta changelog version2](https://user-images.githubusercontent.com/48621967/130053526-2dafa68d-5233-467b-abff-b6fd8812f971.PNG)

I see this as a middle ground, so that it doesn't become overwhelming, but it is also informative.

Let me know what you think